pgjdbc/src/main/java/org/postgresql/core/BaseConnection.java

```diff
@@ -228,4 +232,227 @@ CachedQuery createQuery(String sql, boolean escapeProcessing, boolean isParamete
+   * on this connection. Each request is independent of all other requests
+   * with regard to state local to the connection either on the client or the
+   * server. Work done between {@code beginRequest}, {@code endRequest}
+   * pairs does not depend on any other work done on the connection either as
+   * part of another request or outside of any request. A request may include multiple
+   * transactions. There may be dependencies on committed database state as
+   * that is not local to the connection.
+   * <p>
+   * Local state is defined as any state associated with a Connection that is
+   * local to the current Connection either in the client or the database that
+   * is not transparently reproducible.
+   * <p>
+   * Calls to {@code beginRequest} and {@code endRequest}  are not nested.
+   * Multiple calls to {@code beginRequest} without an intervening call
+   * to {@code endRequest} is not an error. The first {@code beginRequest} call
+   * marks the start of the request and subsequent calls are treated as
+   * a no-op
+   * <p>
+   * Use of {@code beginRequest} and {@code endRequest} is optional, vendor
+   * specific and should largely be transparent. In particular
+   * implementations may detect conditions that indicate dependence on
+   * other work such as an open transaction. It is recommended though not
+   * required that implementations throw a {@code SQLException} if there is an active
+   * transaction and {@code beginRequest} is called.
+   * Using these methods may improve performance or provide other benefits.
+   * Consult your vendors documentation for additional information.
+   * <p>
+   * It is recommended to
+   * enclose each unit of work in {@code beginRequest}, {@code endRequest}
+   * pairs such that there is no open transaction at the beginning or end of
+   * the request and no dependency on local state that crosses request
+   * boundaries. Committed database state is not local.
+   *
+   * @throws SQLException if an error occurs
+   * @implSpec The default implementation is a no-op.
+   * @apiNote This method is to be used by Connection pooling managers.
+   * <p>
+   * The pooling manager should call {@code beginRequest} on the underlying connection
+   * prior to returning a connection to the caller.
+   * <p>
+   * The pooling manager does not need to call {@code beginRequest} if:
+   * <ul>
+   * <li>The connection pool caches {@code PooledConnection} objects</li>
+   * <li>Returns a logical connection handle when {@code getConnection} is
+   * called by the application</li>
+   * <li>The logical {@code Connection} is closed by calling
+   * {@code Connection.close} prior to returning the {@code PooledConnection}
+   * to the cache.</li>
+   * </ul>
+   * @see endRequest
+   * @see PooledConnection
+   * @since 9
+   */
+  @Override
+  default void beginRequest() throws SQLException {
+    throw new SQLFeatureNotSupportedException();
```

Comment:
Make this a noop instead - connection pools won't expect a SQLFeatureNotSupportedException here. IFF supported, this will be called for every transaction.

Reply:
k, thx